### PR TITLE
Cirrus: Use imgts container to record metadata (needs #2213)

### PR DESCRIPTION
**Pull request by cevich**
Make use of the built imgts container image to track
VM image usage statistics for every automation run.

***. DEPENDS ON #2213 .***

Signed-off-by: Chris Evich <cevich@redhat.com>

### CODE REVIEW

**Comment by cevich** [comment]
@baude @mheon This is a followup to #2213 and also depends on ``quay.io/libpod/imgts:latest`` being built: *Until then, the 'meta' test (here) will fail.*

However I did [test the workflow using an image from my registry.](https://cirrus-ci.com/task/6249203488522240).  I can see the updated metadata (labels):

```
[cevich@cevich libpod]$ pgcloud --configuration=libpod compute images describe fedora-28-libpod-9afa57a9 | grep -a3 'labels'
id: '3811958042047208175'
kind: compute#image
labelFingerprint: nD4kG5Ay78k=
labels:
  build-id: '5524526675787776'
  last-used: '1548345695'
  repo-ref: 38d305263720ad9bd964eb690eccf77ad6ec5a44
```

The new 'meta' task can handle some [failures as well, such as a missing or bad image name.](https://cirrus-ci.com/task/5979181209878528)  Exiting successfully despite one failed update.

We don't want random joe-schmoe modifying or clobbering our VM images to do evil things.  So to prove I [got the permissions correct, I tried adding a command to delete an image.](https://cirrus-ci.com/task/6193330527404032)  Updates were successful, and the image-delete failed as expected.

Lastly, I added a small validation check to the gating job.  This will make sure the list of images used for testing are also always labeled with metadata.  I tested the new gating image in the same way and it seems to run cleanly.

**Comment by cevich** [comment]
(rebase & force-push)

**Comment by rhatdan** [comment]
Since #2213 is merged, are we ready to merge this?

**Comment by cevich** [comment]
@rhatdan yes, good to go (IMHO).  I manually confirmed the CI run actually updated metadata (on the VM images in our cloud).

**Comment by rhatdan** [comment]
LGTM
@edsantiago @baude PTAL

**Comment by cevich** [comment]
**note to self**: I should rebase and re-run the tests here to confirm they're all still working.

**Comment by cevich** [comment]
(rebased and --force pushed)

**Comment by cevich** [comment]
(rebased + force pushed)

**Comment by cevich** [comment]
(rebase & force push)

**Comment by rhatdan** [comment]
@cevich Looks like tests are passing.  @edsantiago PTAL

**Comment by edsantiago** [comment]
I've skimmed this a few times now but don't feel qualified to lgtm or even comment: I don't have the necessary Cirrus knowledge.

**Comment by rhatdan** [comment]
@edsantiago None of us other then @cevich do.  :^)

**Comment by cevich** [comment]
@rhatdan I hope that's not entirely true.  Brent's skimmed through things, giusseppe's implemented a new check, and either Adrian or Valentine (can't remember) did some fixup work on CRIU.  It's big to be sure, but I've tried my best to adopt POLA best I can.

@edsantiago It's simpler than travis and maybe even than papr.  The [entire ``.cirrus.yml`` documentation fits on one page](https://cirrus-ci.org/guide/writing-tasks/).  I've also [documented an overview of each task](https://github.com/containers/libpod/tree/master/contrib/cirrus).

*Bottom line:* If there's something not understandable or confusing, please let me know so I can document it :smile:

**Comment by rhatdan** [comment]
/approve
I think we can merge after v1.1

**Comment by cevich** [comment]
rebase and force push

**Comment by cevich** [comment]
@baude @mheon could please take a quick peek, this PR has been waiting nice and patiently :smiley:   This is needed to help support easier VM image-building and management workflows.

**Comment by mheon** [comment]
LGTM

**Comment by mheon** [comment]
Since @rhatdan already LGTM'd
/lgtm

**Comment by cevich** [comment]
Thx @mheon
